src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl.xaml.cs

```diff
@@ -0,0 +1,330 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.OLE.Interop;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Newtonsoft.Json.Linq;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices
+{
+    /// <summary>
+    /// Interaction logic for DocumentOutlineControl.xaml
+    /// </summary>
+    internal partial class DocumentOutlineControl : UserControl, IOleCommandTarget
+    {
+        [MemberNotNullWhen(true, nameof(SymbolsTreeItemsSource))]
+        private bool SymbolTreeInitialized { get; set; }
+        private List<DocumentSymbolViewModel>? SymbolsTreeItemsSource { get; set; }
+
+        private ITextSnapshot? Snapshot { get; set; }
+        private IWpfTextView TextView { get; set; }
+
+        private readonly string? _filePath;
+        private ResettableDelay? _delay;
+        private readonly IAsynchronousOperationListener _asyncListener;
+
+        public DocumentOutlineControl(IWpfTextView textView, ITextBuffer textBuffer, ILanguageServiceBroker2 languageServiceBroker, IThreadingContext threadingContext, IAsynchronousOperationListener asyncListener)
```

Comment:
nit: wrap long lines (good rule of thumb is if they can fit on github screen width).